### ISSUE TITLE
Cannot delete mysql databases with replica instance

### ISSUE THREAD

**Issue by dominikmueller** [GoogleCloudPlatform/gcp-service-broker#466]
### Issue
We are facing the issue that deleting a CloudSQL MySQL service with an added replica instance is throwing an error:

```bash
Really delete the service mysql-ha?> Y
Deleting service mysql-ha in org PENG / space services-test as Dominik.Mueller@<company>...
FAILED
Server error, status code: 502, error code: 10001, message: Service instance mysql-ha: Service broker error: Error deleting instance: googleapi: Error 400: The requested operation is not valid for a replication master instance., errorReplicationMasterInvalidOperation
```

### Steps to reproduce
1. Create a MySQL service instance with a failover replica instance
```bash
cf create-service google-cloudsql-mysql mysql-db-f1-micro mysql-ha \
  -c '{"failover_replica_name": "mysql-ha-replica", "binlog": "true", "backups_enabled": "true"}'
```
2. Delete the created service
```bash
cf delete-service mysql-ha
```

### Root cause
We found that one needs to delete the replica instance before deleting the master instance.
We have been able to confirm this by testing the steps via the gcloud-cli.

After running the following gcloud command, we have been able to delete the service via `cf delete-service`.

```bash
$ gcloud sql instances delete mysql-ha-replica
All of the instance data will be lost when the instance is deleted.
Do you want to continue (Y/n)?  Y

Deleting Cloud SQL instance...done.

Deleted [https://www.googleapis.com/sql/v1beta4/projects/<PROJECT_ID>/instances/mysql-ha-replica].
```

**Comment by josephlewis42** [comment]
Oh, boy. CloudSQL recently updated their API and broke several things the broker was relying on. We have a ticket open with them internally, but I'll attempt to write a work-around for this one too that will delete the replica before the main database and release it with the other fix I have going out soon for #463 .

**Comment by dominikmueller** [comment]
Sounds good, thanks for your effort!
